qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -215,6 +216,21 @@ def load_account(self) -> Optional[AccountProvider]:
 
         return default_provider
 
+    def load_providers(self,
+                       hub: Optional[str] = None,
+                       group: Optional[str] = None,
+                       project: Optional[str] = None) -> None:
+        """Loads a provider by its id (hub/group/project)
+
+        Args:
+            hub: Name of the hub to use.
+            group: Name of the group to use.
+            project: Name of the project to use.
+
+        """
+
+        self._initialize_provider(self._credentials, hub, group, project)
```

Comment:
So every time a user gets a provider this method calls `user_hubs()` again? That seems even less efficient than before where `user_hubs()` was only called once.

Reply:
Fixed via putting user hubs called in account loading/enabling functions so it only executes once